HIP/hip-583.md

```diff
@@ -2,221 +2,268 @@
+2. `key alias` - This format is created by using the bytes of a simple cryptographic public key supported by Hedera
+   (such as `ED25519` or `ECDSA secp256k1`). It is not necessary for the account's key to match the `alias`, only that
+   the account owner has asserted cryptographic proof of ownership of the account `alias` key.
+3. `evm address alias` - This format is created by using the rightmost 20 bytes of the 32 byte `Keccak-256` hash of a
+   `ECDSA secp256k1` public key. This calculation is in the manner described by the Ethereum Yellow Paper.
+   Note, that the recovery id is not formally part of the public key and not included in the hash. This is
+   calculated on the consensus nodes using the `ECDSA` key provided in the auto create `CryptoTransfer` transaction.
+
+Examples of each format are noted below:
+(Please NOTE: the `alias` is a binary format when encoded on-wire for protobuf. We represent the bytes as base-32
+encoded ASCII for clarity in the documentation, but in reality these are raw bytes)
+
+| Identifier Type                | Example                                                      |
+|--------------------------------|--------------------------------------------------------------|
+| Hedera Account ID              | 0.0.10                                                       |
+| Hedera Account Long-Zero Alias | 0.0.000000000000000000000000000000000000000a                 |
+| Hedera Key Alias               | 0.0.CIQNOWUYAGBLCCVX2VF75U6JMQDTUDXBOLZ5VJRDEWXQEGTI64DVCGQ  |
+| EVM Address Alias              | 0.0.b794f5ea0ba39494ce839613fffba74279579268                 |
+
+TODO: Are all three of these base-32 encoded? Should use consistent encoding
+
+To provide greater context for this HIP some additional terms useful:
+
+- `Ethereum-native wallets` - wallets that utilize the Ethereum JSON-RPC APIs implemented by the JSON-RPC Relay.
+  Accounts created via these wallets are `ECDSA` derived keys and use the `evm address alias` format.
+- `Hedera-native wallets` - wallets that utilize the Hedera SDK or gRPC HAPI calls to interact with Hedera consensus
+  and mirror nodes. Historically, accounts created via these wallets are `ED25519` based and usually identify the
+  account via the `Hedera Account Id`.
+- `auto-create` - a unique account creation flow in which the `key alias` or `evm address alias` that a crypto-transfer
+  is addressed to is used to automatically create a new account (if none exists) without an explicit `CryptoCreate`
+  transaction being submitted.
+- `Hollow account` - a partially created account which lacks a public key but does have a valid `shard.realm.num`
+  accountId _and_ is auto-created with an `evm address alias`. Such an account may receive assets but may not take
+  part in transactions that require its signature, until it is promoted to being a "full" account.
+- `lazy-create` - a unique account creation flow in which the public address (extracted from a provided ECDSA public
```

Comment:
True. From a linguistics perspective I think of auto-create being used to create hollow accounts or full accounts.

Reply:
However, "lazy create" as a way of capturing "auto-create with evm address alias leading to hollow account" is a useful shortcut, otherwise I have to write all of that every time.

Reply:
I don't think `evm address alias` should be added to the `auto crate` definition as an `auto-create` should produce a full account not a hollow account.